### PR TITLE
Include admin resource requests with support for allocation

### PR DESCRIPTION
**Pull request by cdosborn**
## Description

This is the pull request that the new resource requests depends on. I will link the tropo-pr shortly.

## Checklist before merging Pull Requests
- [ ] Finish review step in [jira](https://pods.iplantcollaborative.org/jira/browse/ATMO-1912#comment-59495)
- [ ] New test(s) included to reproduce the bug/verify the feature
- [ ] Documentation created/updated at [Example link to documentation](https://example.test/doc#new_section) to give context to the feature
- [ ] Reviewed and approved by at least one other contributor.
- [ ] If necessary, include a snippet in CHANGELOG.md

### CODE REVIEW

File: api/v2/views/allocation_source.py [hunk 1]
```diff
@@ -26,11 +26,23 @@ class AllocationSourceViewSet(MultipleFieldLookup, AuthModelViewSet):
 
     def get_queryset(self):
         """
-        Get allocation sources for user
+        Get allocation sources by user or by all users
         """
         user = self.request.user
-        sources = UserAllocationSource.objects.filter(user=user).values_list('allocation_source')
-        return AllocationSource.objects.filter(id__in=sources)
+        if user.is_admin:
```
Comment (steve-gregory) [inline api/v2/views/allocation_source.py]: `is_admin` is not a property, it is a function:

https://github.com/cyverse/atmosphere/blob/zesty-zapdos/core/models/user.py#L89-L92

We should either change this to `is_admin()` or add the `@property` decorator over `is_admin`

Comment (cdosborn) [inline api/v2/views/allocation_source.py]: Nice

Comment (cdosborn) [inline api/v2/views/allocation_source.py]: Thats going to be in a few places.

Comment (cdosborn) [inline api/v2/views/allocation_source.py]: The respective commits have been updated.

File: api/v2/serializers/details/resource_request.py [hunk 2]
```diff
@@ -1,177 +1,20 @@
...
-    ProviderSummarySerializer,
-    QuotaSummarySerializer,
-    StatusTypeSummarySerializer
+from rest_framework import serializers
+from core.models import ResourceRequest
+from api.v2.serializers.summaries import UserSummarySerializer
+from api.v2.serializers.fields import (
+    IdentityRelatedField, StatusTypeRelatedField
 )
 from api.v2.serializers.fields.base import UUIDHyperlinkedIdentityField
```
Comment (steve-gregory) [inline api/v2/serializers/details/resource_request.py]: 👍  Thanks for cleaning up this file

File: api/v2/serializers/details/resource_request.py [hunk 3]
```diff
@@ -183,56 +26,13 @@ class Meta:
...
-            'current_allocation'
         )
 
 
-class UserResourceRequestSerializer(serializers.HyperlinkedModelSerializer):
+class UserResourceRequestSerializer(ResourceRequestSerializer):
     def validate_status(self, value):
         if str(value) not in ["pending", "closed"]:
             raise serializers.ValidationError("Users can only open and close requests.")
         return value
```
Comment (steve-gregory) [inline api/v2/serializers/details/resource_request.py]: 👍 ♻️

**Comment by cdosborn** [comment]
Let me restart the build before this gets merged to check migrations/etc.

File: api/v2/views/identity.py [hunk 4]
```diff
@@ -56,27 +59,34 @@ def get_queryset(self):
...
-        data = request.data
-        if 'quota' in data and 'id' in data.get('quota'):
-            quota_id = data.get('quota').get('id')
-            identity.quota = Quota.objects.get(id=quota_id)
-            identity.save()
-            admin_task.set_provider_quota.delay(str(identity.uuid))
-
-        serialized = serializer(identity,
-                                context={'request': self.request})
+        SerializerCls = self.get_serializer_class()
```
Comment (cdosborn) [inline api/v2/views/identity.py]: 👍

File: core/models/event_table.py [hunk 5]
```diff
@@ -23,6 +23,7 @@
     listen_for_allocation_source_removed,
     listen_for_instance_allocation_removed
 )
+from core.hooks.quota import listen_for_quota_assigned
```
Comment (cdosborn) [inline core/models/event_table.py]: unused import?

**Comment by cdosborn** [comment]
This is ready!

**Comment by steve-gregory** [comment]
Thanks @cdosborn :shipit:

**Pull request merged**